clusterloader2/pkg/measurement/common/scheduling_throughput_prometheus.go

```diff
@@ -0,0 +1,102 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	schedulingThroughputPrometheusMeasurementName = "SchedulingThroughputPrometheus"
+
+	maxSchedulingThroughputQuery = `max_over_time(sum(irate(apiserver_request_total{verb="POST", resource="pods", subresource="binding",code="201"}[1m]))[%v:5s])`
+)
+
+func init() {
+	create := func() measurement.Measurement { return CreatePrometheusMeasurement(&schedulingThroughputGatherer{}) }
+	if err := measurement.Register(schedulingThroughputPrometheusMeasurementName, create); err != nil {
+		klog.Fatalf("Cannot register %s: %v", schedulingThroughputMeasurementName, err)
+	}
+}
+
+type schedulingThroughputGatherer struct{}
+
+func (a *schedulingThroughputGatherer) Gather(executor QueryExecutor, startTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
+	throughputSummary, err := a.getThroughputSummary(executor, startTime, config)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := util.PrettyPrintJSON(throughputSummary)
+	if err != nil {
+		return nil, err
+	}
+
+	threshold, err := util.GetFloat64OrDefault(config.Params, "threshold", 0)
+	if threshold > 0 && throughputSummary.Max < threshold {
+		err = errors.NewMetricViolationError(
+			"scheduler throughput_prometheus",
+			fmt.Sprintf("actual throughput %f lower than threshold %f", throughputSummary.Max, threshold))
+	}
+
+	summaries := []measurement.Summary{
+		measurement.CreateSummary(a.String(), "json", content),
+	}
+
+	return summaries, err
+}
+
+func (a *schedulingThroughputGatherer) getThroughputSummary(executor QueryExecutor, startTime time.Time, config *measurement.Config) (*schedulingThroughputPrometheus, error) {
+	measurementEnd := time.Now()
+	measurementDuration := measurementEnd.Sub(startTime)
+	promDuration := measurementutil.ToPrometheusTime(measurementDuration)
+	query := fmt.Sprintf(maxSchedulingThroughputQuery, promDuration)
+
+	samples, err := executor.Query(query, measurementEnd)
+	if err != nil {
+		return nil, err
+	}
+	if len(samples) != 1 {
+		return nil, fmt.Errorf("got unexpected number of samples: %d", len(samples))
+	}
+
+	maxSchedulingThroughput := samples[0].Value
+	throughputSummary := &schedulingThroughputPrometheus{
+		Max: float64(maxSchedulingThroughput),
+	}
+
+	return throughputSummary, nil
+}
+
+func (a *schedulingThroughputGatherer) String() string {
+	return schedulingThroughputPrometheusMeasurementName
+}
+
+func (a *schedulingThroughputGatherer) IsEnabled(config *measurement.Config) bool {
+	return true
+}
+
+type schedulingThroughputPrometheus struct {
```

Comment:
This requires changes to perfdash: https://github.com/kubernetes/perf-tests/blob/28d4b32e0d2cb9b5b94a8769286bfd0f0ca771d8/perfdash/parser.go#L287-L309

I think we can submit this, and fix perfdash in next iteration.